pkg/platform/runtime/model/buildplanner.go

```diff
@@ -227,40 +227,46 @@ func removeEmptyTargets(bp *model.BuildPlan) {
 }
 
 type StageCommitParams struct {
-	Owner            string
-	Project          string
-	ParentCommit     string
+	Owner        string
+	Project      string
+	ParentCommit string
+	// Commits can have either an operation (e.g. installing a package)...
 	PackageName      string
 	PackageVersion   string
 	PackageNamespace vcsModel.Namespace
 	Operation        model.Operation
+	// ... or a script (e.g. from pull).
+	Script *model.BuildExpression
```

Comment:
I believe this comment says that when we run `state pull` and the script needs to be updated we will end up submitting that to the BuildPlanner rather than the traditional commit operation parameters. Is that correct? 

Either way, I think we should make this comment a bit more clear. It took me a second to follow so someone else who's unfamiliar might benefit from something more detailed.

Reply:
Yes, that is correct. I've updated the comment. Hopefully it's more helpful. Feel free to suggest something else if it's not.